ocp_utilities/infra.py

```diff
@@ -114,6 +117,69 @@ def assert_pods_failed_or_pending(pods):
         )
 
 
+def assert_nodes_in_healthy_condition(
+    nodes,
+    healthy_node_condition_type=None,
+):
+    """
+    Validates nodes are in a healthy condition.
+    Nodes Ready condition is True and none of the following node conditions is True:
+        - DiskPressure
+        - MemoryPressure
+        - PIDPressure
+        - NetworkUnavailable
+        - OutOfDisk
+
+    Args:
+         nodes(list): List of Node objects
+
+         healthy_node_condition_type (dict):
+            Dictionary with condition type and the respective healthy condition
+                status: Example: {"DiskPressure": "False", ...}
+
+    Raises:
+        NodesNotHealthyConditionError: if any nodes DiskPressure MemoryPressure,
+            PIDPressure, NetworkUnavailable, etc condition is True
+    """
+    LOGGER.info("Verify all nodes are in a healthy condition.")
+
+    if not healthy_node_condition_type:
+        healthy_node_condition_type = {
+            "OutOfDisk": Node.Condition.Status.FALSE,
+            "DiskPressure": Node.Condition.Status.FALSE,
+            "MemoryPressure": Node.Condition.Status.FALSE,
+            "NetworkUnavailable": Node.Condition.Status.FALSE,
+            "PIDPressure": Node.Condition.Status.FALSE,
+            Node.Condition.READY: Node.Condition.Status.TRUE,
+        }
+
+    elif type(healthy_node_condition_type) != dict:
+        raise TypeError(
+            f"A dict is required but got type {type(healthy_node_condition_type)}"
+        )
+
+    unhealthy_nodes_with_conditions = {}
+    for node in nodes:
+        unhealthy_condition_type_list = [
+            condition.type
+            for condition in node.instance.status.conditions
+            if condition.type in healthy_node_condition_type
+            and healthy_node_condition_type[condition.type] != condition.status
```

Comment:
`healthy_node_condition_type[condition.type]` will fail if the key does not exist

Reply:
@rnetser since there is the first check `if condition.type in healthy_node_condition_type`, it won't get to the second check (`healthy_node_condition_type[condition.type] != condition.status`), if the first check is false.